event-svc/src/event/service.rs

```diff
@@ -145,15 +149,18 @@ impl EventService {
     /// In the future, we will need to do more event validation (verify all EventID pieces, hashes, signatures, etc).
     pub(crate) async fn parse_discovered_event(
         item: &ReconItem<EventId>,
+        source: Option<String>,
```

Comment:
I'm wondering if it might make sense to introduce a new type, even if it's just an alias of String, something like `DIDString`, just so it's clearer when reading the code what is being passed around here and various other places that take a DID string?  WDYT @dav1do?

Reply:
or a `[u8; 32]` NodeID class that can to and from `Cid`, `PeerId`, and `DidKey`

```rs
pub struct NodeId {
    public_ed25519_key_bytes: [u8; 32],
    singing_ed25519_keypair: Option<Ed25519KeyPair>,
}

impl NodeId {
    pub fn cid(&self) -> Cid {todo!()}
    pub fn did_key(&self) -> String {todo!()}
    pub fn ed25519_key_pair(&self) -> Ed25519KeyPair {todo!()}
    pub fn peer_id(&self) -> PeerId {todo!()}
    pub fn try_from_cid(cid: Cid)  -> Result<Self> {todo!()}
    pub fn try_from_did_key(did_key: &str) -> Result<Self> {todo!()}
    pub fn try_from_dir(key_dir: PathBuf) -> Result<Self> {todo!()}
    pub fn try_from_secret(secret: &str) -> Result<Self> {todo!()}
    pub fn try_from(peer_id: PeerId) -> Result<Self> {todo!()}
}
```

Reply:
they are NodeId now